pkg/api/submit.proto

```diff
@@ -173,6 +184,12 @@ service Submit {
             body: "*"
         };
     }
+    rpc CancelJobSet (JobSetCancelRequest) returns (CancellationResult) {
```

Comment:
Why do we need a new API for this?  

Wouldn't it be better to just add JobSetFilter to the JobCancelRequest and utilize CancelJobs rpc call for this?

Reply:
I think it's something we want to discuss internally. For a while we've wanted to split that endpoint rather than having a combined endpoint (cancel by job and cancel by jobset).

We absolutely could continue to extend the existing and the only reason I didn't, is to move towards what we want in future. 

If we decide we don't want that, I'll swap to extending the existing as you suggested.

Reply:
I like that.  I have been meaing to create a python functions that call the CancelJob via jobset/queue or job since the behavior is different and this saves me that work.  

So our goal would be to have CancelJobs take only a jobId and CancelJobSet to take queue and JobSet and list of states?

We should file that as an issue so we document that this is something we'd like.  It is a good issue for anyone to tackle too!